go/vt/vtgate/evalengine/testcases/inputs.go

```diff
@@ -97,6 +97,12 @@ var inputConversions = []string{
 	"18446744073709540000e0",
 	"-18446744073709540000e0",
 	"JSON_OBJECT()", "JSON_ARRAY()",
+	"cast(0 as json)", "cast(1 as json)",
+	"cast(true as json)", "cast(false as json)",
+	"cast('{}' as json)", "cast('[]' as json)",
+	"cast('null' as json)", "cast('true' as json)", "cast('false' as json)",
+	"cast('1' as json)", "cast('1.1' as json)", "cast('-1.1' as json)",
+	"cast('\"foo\"' as json)", "cast('invalid' as json)",
```

Comment:
These all test various JSON edge cases in casting and operations that drive a bunch of the changes here.